### ISSUE TITLE
Case of Targeted attack

### ISSUE THREAD

**Issue by abhishekaich27** [Muzammal-Naseer/CDA#8]
For the case of a targeted attack, do we train the generator only for one target, indicated by the argument passed?

**Comment by Muzammal-Naseer** [comment]
We trained generator only for one epoch on ImageNet trainset and for 10 epochs on painting and comics.

**Comment by abhishekaich27** [comment]
Oh, apologies for the poor wordings! I meant how do you set the targeted attack training for any dataset? For the untargeted attack, I understand it's -1 to indicate that the generator should learn to misclassify whatever the surrogate model says about the input clean image. But what about the case of a targeted attack? Do you set just 1 target for all the classes?

**Comment by Muzammal-Naseer** [comment]
Hi...its alright.

I have updated the "train.py" for targeted training using relativistic cross entropy. 

Now if you want to let say train targeted generator for "Goose (target_label 99, in ImageNet)" then just set --target 99. Generator will be optimized to project samples from a given domain such that they are misclassified as Goose. 

Yes, one generator for one class. This is a limitation of our method in the case of targetted attack. An interesting research direction would be to explore if a single generator can be trained for multiple targetted attacks.
 
Please feel free to ask further questions.

**Comment by abhishekaich27** [comment]
Thank you so much.  So, correct me if I am wrong. Basically, we can only train the generator for 1 label (as a targeted attack) at a time. For every targeted attack with a **new** label, we need to train the generator again.

**Comment by Muzammal-Naseer** [comment]
Yes, that's correct.
